src/rules/dynamic-import-chunkname.js

```diff
@@ -92,12 +94,24 @@ module.exports = {
           return;
         }
 
+        if (eagerModeRegex.test(comment.value)) {
+          isEagerModePresent = true;
+        }
+
         if (chunkSubstrRegex.test(comment.value)) {
           isChunknamePresent = true;
         }
       }
 
-      if (!isChunknamePresent && !allowEmpty) {
+      if (isChunknamePresent && isEagerModePresent) {
+        context.report({
+          node,
+          message:
+            `dynamic imports should not have both webpackChunkName and webpackMode: "eager"`,
+        });
+      }
```

Comment:
```suggestion
      if (isChunknamePresent && isEagerModePresent) {
        context.report({
          node,
          message: 'dynamic imports should not have both webpackChunkName and webpackMode: "eager"`\',
        });
      }
```

also is there any reason why someone might still want a chunk name?

Reply:
I don't think having a chunk name with eager mode makes sense, since eager mode explicitly disables creating a new chunk, so the module is just lazily evaluated instead of lazily loaded+evaluated. Adding a chunk name anyways basically reverts back to being `webpackMode: 'lazy'` and generates a new chunk. A warning would help users catch unintentional additional chunks when using `webpackMode: 'eager'`

Reply:
ok, so having both is always a mistake. thanks!

Reply:
altho actually - that means that removing the name would quietly change the mode from lazy to eager - i think this should be a suggestion to *either* remove the name, *or* remove/change the webpackMode, to be safe.